lib/markdown_videos/services/service_base.rb

```diff
@@ -0,0 +1,124 @@
+module MarkdownVideos
+  module Services
+
+    class ServiceBase
+
+      attr_accessor :id,
+                    :alt_text,
+                    :markdown_url,
+                    :width,
+                    :height,
+                    :wrapper,
+                    :classname
+
+      def self.support(markdown_url)
+        markdown_url =~ regexp
+      end
+
+      def self.regexp
+        raise "#{self.class} class should respond to `regexp`"
```

Comment:
What about raising the NotImplementedError error? And then changing the message to be more like `"#{self.class} should implement the #regexp method"` or something,